fixlib/fixlib/basecategories.py

```diff
@@ -0,0 +1,113 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any
+
+
+@dataclass(frozen=True)
+class BaseCategory:
```

Comment:
eq should be implemented by dataclass automatically.
You could also add @functools.total_ordering to the class and only implement __lt__.

Reply:
> __eq__ should be implemented by dataclass automatically.

It is, but it's not based on comparing str(self).